src/sonic-dhcp-server/dhcp_server/common/utils.py

```diff
@@ -93,6 +97,43 @@ def merge_intervals(intervals):
     return ret
 
 
+def validate_str_type(type, value):
+    """
+    To validate whether type is consistent with string value
+    Args:
+        type: string, value type
+        value: checked value
+    Returns:
+        True, type consistent with value
+        False, type not consistent with value
+    """
+    if not isinstance(value, str):
+        return False
+    if type not in SUPPORT_TYPE:
+        syslog.syslog(syslog.LOG_ERR, "Unsupported validate type: {}".format(type))
+        return False
+    if type == "string":
+        return True
+    if type == "binary":
+        if len(value) == 0 or len(value) % 2 != 0:
+            return False
+        return all(c in set(string.hexdigits) for c in value)
+    if type == "boolean":
+        return value in ["true", "false", "0", "1"]
```

Comment:
Do we need support "0" or "1"? We can add more restrict condition to avoid extra complexity